prow/config/jobs.go

```diff
@@ -498,6 +498,11 @@ func (ps Presubmit) TriggersConditionally() bool {
 	return ps.NeedsExplicitTrigger() || ps.RegexpChangeMatcher.CouldRun()
 }
 
+// RequiresManualTrigger determines if the presubmit requires manual trigger and do not auto-run, even conditionally.
+func (ps Presubmit) RequiresManualTrigger() bool {
+	return !ps.AlwaysRun && ps.SkipIfOnlyChanged == "" && ps.RunIfChanged == ""
+}
```

Comment:
I think this is equivalent to `NeedsExplicitTrigger` - `CouldRun` is true when at least one is set and false when neither is.

```go
func (ps Presubmit) NeedsExplicitTrigger() bool {
	return !ps.AlwaysRun && !ps.RegexpChangeMatcher.CouldRun()
}
...
// CouldRun determines if its possible for a set of changes to trigger this condition
func (cm RegexpChangeMatcher) CouldRun() bool {
	return cm.RunIfChanged != "" || cm.SkipIfOnlyChanged != ""
}
```

Reply:
Yeah, but I guess we need  `return !ps.AlwaysRun && ps.SkipIfOnlyChanged == "" && ps.RunIfChanged == ""` to be the only case in which we want to enable this. Your solution gives also other resolutions (when one of changed is true). Is that what we want?

Reply:
No, if you inline the `CouldRun` into `NeedsExplicitTrigger` you can see that it is equivalent to your new `RequiresManualTrigger` (demorgan's):

```
ps.SkipIfOnlyChanged == "" && ps.RunIfChanged == ""
<=>
!(ps.RegexpChangeMatcher.RunIfChanged != "" || ps.RegexpChangeMatcher.SkipIfOnlyChanged != "")
```

```
// RequiresManualTrigger determines if the presubmit requires manual trigger and do not auto-run, even conditionally.
func (ps Presubmit) RequiresManualTrigger() bool {
	return !ps.AlwaysRun && ps.SkipIfOnlyChanged == "" && ps.RunIfChanged == ""
}

// NeedsExplicitTrigger determines if the presubmit requires a human action to trigger it or not.
func (ps Presubmit) NeedsExplicitTrigger() bool {
	return !ps.AlwaysRun && !(ps.RegexpChangeMatcher.RunIfChanged != "" || ps.RegexpChangeMatcher.SkipIfOnlyChanged != "")
}
```

Reply:
Ah, I have not noticed one ! in the NeedsExplicitTrigger